src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -156,9 +118,23 @@ main() {
   aws_login
   aws_relogin_loop & AWS_RELOGIN_PID=$!
 
+  set_vars_session_and_cache
```

Comment:
Consolidated preserve_cache-related setup here, and moved some of it to helper functions.